pkg/validation/application_installation_test.go

```diff
@@ -152,13 +183,49 @@ func TestValidateApplicationInstallationUpdate(t *testing.T) {
 					return *spec
 				}(),
 			},
-			expectedError: `[spec.applicationRef.name: Invalid value: "app": field is immutable]`,
+			expectedError: `[spec.applicationRef.name: Invalid value: "updated-app": field is immutable]`,
+		},
+		{
+			name: "Update ApplicationInstallation Success - ReconciliationInterval equals 0",
+			ai:   ai,
+			updatedAI: &appskubermaticv1.ApplicationInstallation{
+				Spec: func() appskubermaticv1.ApplicationInstallationSpec {
+					spec := ai.Spec.DeepCopy()
+					spec.ReconciliationInterval = metav1.Duration{Duration: 0}
+					return *spec
+				}(),
+			},
+			expectedError: "[]",
+		},
+		{
+			name: "Update ApplicationInstallation Success - ReconciliationInterval greater than 0",
+			ai:   ai,
+			updatedAI: &appskubermaticv1.ApplicationInstallation{
+				Spec: func() appskubermaticv1.ApplicationInstallationSpec {
+					spec := ai.Spec.DeepCopy()
+					spec.ReconciliationInterval = metav1.Duration{Duration: 10 * time.Minute}
+					return *spec
+				}(),
+			},
+			expectedError: "[]",
+		},
+		{
+			name: "Update ApplicationInstallation Failure - Invalid ReconciliationInterval less than 0",
+			ai:   ai,
+			updatedAI: &appskubermaticv1.ApplicationInstallation{
+				Spec: func() appskubermaticv1.ApplicationInstallationSpec {
+					spec := ai.Spec.DeepCopy()
+					spec.ReconciliationInterval = metav1.Duration{Duration: -10}
+					return *spec
+				}(),
+			},
+			expectedError: `[spec.reconciliationInterval: Invalid value: "-10ns": should be a positive value, or zero to disable]`,
 		},
 	}
 
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
-			err := ValidateApplicationInstallationUpdate(context.Background(), fakeClient, *testCase.ai, *testCase.updatedAI)
+			err := ValidateApplicationInstallationUpdate(context.Background(), fakeClient, *testCase.updatedAI, *testCase.ai)
```

Comment:
Note: parameters were inverted but we didn't notice because we were testing the immutability of fields.